frontend/ui/impl/src/main/java/com/virtuslab/gitmachete/frontend/ui/impl/table/DemoGitMacheteRepositorySnapshot.java

```diff
@@ -133,13 +133,18 @@ public Option<IExecutionResult> executeMachetePreRebaseHookIfPresent(IGitRebaseP
   }
 
   @Override
-  public OngoingRepositoryOperation getOngoingRepositoryOperation() {
-    return OngoingRepositoryOperation.NO_OPERATION;
-  }
-
-  @Override
-  public Option<String> getOngoingOperationsBaseBranchName() {
-    return Option.none();
+  public IOngoingRepositoryOperationInfo getOngoingRepositoryOperationInfo() {
+    return new IOngoingRepositoryOperationInfo() {
```

Comment:
Also, I'd name the data class `OngoingRepositoryOperation`, and the enum `OngoingRepositoryOperationType`